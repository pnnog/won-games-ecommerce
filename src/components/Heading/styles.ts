import styled, { DefaultTheme, css } from 'styled-components';
import media from 'styled-media-query';
import { HeadingProps, LineColorProps } from '.';

const wrapperModifiers = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,

  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};

    ${media.greaterThan('medium')`
    font-size: ${theme.font.sizes.xxlarge};
    
    `}
  `,

  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.huge};
  `,

  leftLine: (theme: DefaultTheme, lineColor: LineColorProps) => css`
    padding-left: ${theme.spacings.xxsmall};
    border-left: 0.7rem solid ${theme.colors[lineColor]};
  `,

  bottomLine: (theme: DefaultTheme, lineColor: LineColorProps) => css`
    position: relative;
    margin-bottom: ${theme.spacings.medium};

    &::after {
      position: absolute;
      left: 0;
      bottom: -1rem;
      content: '';
      border-bottom: 0.5rem solid ${theme.colors[lineColor]};
      width: 5rem;
    }
  `,
};

export const Wrapper = styled.h2<HeadingProps>`
  ${({ theme, color, size, leftLine, bottomLine, lineColor }) => css`
    color: ${theme.colors[color!]};

    ${size && wrapperModifiers[size](theme)}

    ${!!leftLine && lineColor && wrapperModifiers.leftLine(theme, lineColor!)}

    ${!!bottomLine &&
    lineColor &&
    wrapperModifiers.bottomLine(theme, lineColor!)}
  `}
`;
