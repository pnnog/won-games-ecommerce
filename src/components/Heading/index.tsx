import * as S from './styles';

export type LineColorProps = 'primary' | 'secondary';

export type HeadingProps = {
  children: React.ReactNode;
  color?: 'white' | 'black';
  size?: 'small' | 'medium';
  leftLine?: boolean;
  bottomLine?: boolean;
  lineColor?: LineColorProps;
};

const Heading = ({
  children,
  color = 'white',
  size = 'medium',
  leftLine = false,
  bottomLine = false,
  lineColor = 'primary',
}: HeadingProps) => (
  <S.Wrapper
    color={color}
    size={size}
    leftLine={leftLine}
    bottomLine={bottomLine}
    lineColor={lineColor}
  >
    {children}
  </S.Wrapper>
);

export default Heading;
