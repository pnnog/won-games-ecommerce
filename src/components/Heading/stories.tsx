import { Meta, Story } from '@storybook/react';
import Heading from '.';
import { HeadingProps } from '.';

export default {
  title: 'Heading',
  component: Heading,

  argTypes: {
    children: {
      type: 'string',
    },
  },
} as Meta;

export const Default: Story<HeadingProps> = (args) => <Heading {...args} />;

Default.args = {
  children: 'Most Populars',
};
