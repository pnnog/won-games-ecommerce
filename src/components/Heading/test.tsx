import { screen } from '@testing-library/react';
import { renderWithTheme } from '../../utils/tests/helpers';

import Heading from '.';

describe('<Heading />', () => {
  it('should render white heading by default ', () => {
    renderWithTheme(<Heading> Renderizando Teste</Heading>);
    expect(
      screen.getByRole('heading', { name: /Renderizando teste/i })
    ).toHaveStyle({ color: '#FAFAFA' });
  });

  it('should render black heading when color is passed ', () => {
    renderWithTheme(<Heading color="black"> Renderizando Teste</Heading>);
    expect(
      screen.getByRole('heading', { name: /Renderizando teste/i })
    ).toHaveStyle({ color: '#030517' });
  });

  it('should render heading with leftLine ', () => {
    renderWithTheme(<Heading leftLine={true}> Renderizando Teste</Heading>);
    expect(
      screen.getByRole('heading', { name: /Renderizando teste/i })
    ).toHaveStyle({ 'padding-left': '0.8rem' });
  });

  it('should render heading with bottomLine ', () => {
    renderWithTheme(<Heading bottomLine={true}> Renderizando Teste</Heading>);
    expect(
      screen.getByRole('heading', { name: /Renderizando teste/i })
    ).toHaveStyleRule('border-bottom', '0.5rem solid #F231A5', {
      modifier: '::after',
    });
  });
});
