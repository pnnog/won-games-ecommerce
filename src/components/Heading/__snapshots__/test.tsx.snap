// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render heading with huge font size when size is huge 1`] = ` `;
